Riot/Modules/Room/RoomInfo/RoomInfoList/RoomInfoListViewController.swift

```diff
@@ -0,0 +1,407 @@
+            self.viewModel.process(viewAction: .navigate(target: .settings))
+        }
+        let text = viewModel.numberOfMembers == 1 ? VectorL10n.roomInfoListOneMember : VectorL10n.roomInfoListXMembers(String(viewModel.numberOfMembers))
+        let row_2_2 = Row(type: .default, icon: Asset.Images.userIcon.image, text: text, accessoryType: .disclosureIndicator) {
+            self.viewModel.process(viewAction: .navigate(target: .members))
+        }
+        let row_2_3 = Row(type: .default, icon: Asset.Images.scrollup.image, text: VectorL10n.roomDetailsFiles, accessoryType: .disclosureIndicator) {
+            self.viewModel.process(viewAction: .navigate(target: .uploads))
+        }
+        
+        let section2 = Section(header: VectorL10n.roomInfoListSectionOther,
+                               rows: [row_2_0,
+                                      row_2_2,
+                                      row_2_3],
+                               footer: nil)
+        
+        let row_3_0 = Row(type: .destructive, icon: Asset.Images.roomActionLeave.image, text: VectorL10n.roomParticipantsLeavePromptTitle, accessoryType: .none) {
+            self.viewModel.process(viewAction: .leave)
+        }
+        let section3 = Section(header: nil,
+                               rows: [row_3_0],
+                               footer: nil)
+        
+        tmpSections.append(section2)
+        tmpSections.append(section3)
+        
+        sections = tmpSections
+    }
+    
+    private func update(theme: Theme) {
+        self.theme = theme
+        
+        self.view.backgroundColor = theme.headerBackgroundColor
+        self.mainTableView.backgroundColor = theme.headerBackgroundColor
+        
+        if let navigationBar = self.navigationController?.navigationBar {
+            theme.applyStyle(onNavigationBar: navigationBar)
+        }
+
+        closeButton.backgroundColor = theme.headerBorderColor
+        closeButton.tintColor = theme.textSecondaryColor
+        closeButton.setImage(closeButton.image(for: .normal)?.vc_tintedImage(usingColor: theme.textSecondaryColor), for: .normal)
+        
+        mainTableView.reloadData()
+    }
+    
+    private func registerThemeServiceDidChangeThemeNotification() {
+        NotificationCenter.default.addObserver(self, selector: #selector(themeDidChange), name: .themeServiceDidChangeTheme, object: nil)
+    }
+    
+    @objc private func themeDidChange() {
+        self.update(theme: ThemeService.shared().theme)
+    }
+    
+    private func setupViews() {
+        self.navigationItem.rightBarButtonItem = MXKBarButtonItem(customView: closeButton)
+        
+        self.title = ""
+        
+        mainTableView.register(cellType: TextViewTableViewCell.self)
+        mainTableView.register(cellType: RoomInfoBasicTableViewCell.self)
+        mainTableView.register(headerFooterViewType: TableViewHeaderFooterView.self)
```

Comment:
`TableViewHeaderFooterView` name is too generic, we don't guess the purpose of the class, we should rename it at least to something like `TextViewTableHeaderFooterView`. You can also use a nib with `UITableViewHeaderFooterView` if you want, you can have a look to `EditHistoryHeaderView` class for an example.

Reply:
You're right. Updated